data-prepper-plugins/grok-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/grok/GrokProcessor.java

```
@@ -267,7 +267,14 @@ private void compileMatchPatterns() {
         for (final Map.Entry<String, List<String>> entry : grokProcessorConfig.getMatch().entrySet()) {
             fieldToGrok.put(entry.getKey(), entry.getValue()
                             .stream()
-                            .map(item -> grokCompiler.compile(item, grokProcessorConfig.isNamedCapturesOnly()))
+                            .map(item -> {
+                                try {
+                                    return grokCompiler.compile(item, grokProcessorConfig.isNamedCapturesOnly());
+                                } catch (IllegalArgumentException e) {
+                                    throw new RuntimeException(
+                                            String.format("Invalid regex pattern in match.%s", entry.getKey()), e);
```

Do we want to stop the stream processing if we encounter IllegalArugmentException? or we want to collect the errors but continue with stream processing? 

Also, in the exception we are throwing, we are attaching the original `e`. Depending on whoever is handling this exception, it could potentially print the entire stacktrace for each failure and create a lot of noise in the logs.

---

This will be a config validation error which means data prepper will crash at runtime. The change I made is to increase clarity on the failure message:

previous:
```
s3-log-pipeline.processor.grok: caused by: Exception thrown from plugin "grok". caused by: No definition for key 'ses_logs' found, aborting
```
now
```
2025-02-20T13:14:40,784 [main] ERROR org.opensearch.dataprepper.core.validation.LoggingPluginErrorsHandler - 1. waf-access-log-pipeline.processor.grok: caused by: Exception thrown from plugin "grok". caused by: Invalid regex pattern in match.message caused by: No definition for key 'CUSTOM_PATTERN_FROM_FILE' found, aborting
```

---

Got it. It is not a data processing error 👍

---

Just one thought, instead of RuntimeException, we could possibly throw `InvalidPluginConfigurationException` to be more explicit